RefRed/sf_calculator/lr_data.py

```diff
@@ -51,19 +50,19 @@ def __init__(self, workspace, read_options):
         self.full_file_name = mt_run.getProperty('Filename').value[0]
         self.filename = os.path.basename(self.full_file_name)
 
-        sample = self.workspace.getInstrument().getSample()
-        source = self.workspace.getInstrument().getSource()
+        sample = workspace.getInstrument().getSample()
+        source = workspace.getInstrument().getSource()
         self.dMS = sample.getDistance(source)
 
         # create array of distances pixel->sample
-        self.number_x_pixels = int(self.workspace.getInstrument().getNumberParameter("number-of-x-pixels")[0])  # 256
-        self.number_y_pixels = int(self.workspace.getInstrument().getNumberParameter("number-of-y-pixels")[0])
+        self.number_x_pixels = int(workspace.getInstrument().getNumberParameter("number-of-x-pixels")[0])  # 256
+        self.number_y_pixels = int(workspace.getInstrument().getNumberParameter("number-of-y-pixels")[0])
 
         dPS_array = np.zeros((self.number_x_pixels, self.number_y_pixels))
```

Comment:
Same here: only one element out of the whole array is used, so can you get rid of it and the whole `for` loop?